### PR TITLE
Adds `toString()` for the Java generator

### PR DESCRIPTION
**Pull request by steipete**
Similar to #131, implements `toString()` for the Java generator for easier debugging.

### CODE REVIEW

File: src/source/JavaGenerator.scala [hunk 1]
```diff
@@ -340,6 +340,19 @@ class JavaGenerator(spec: Spec) extends Generator(spec) {
...
+        w.wl("@Override")
+        w.w("public String toString()").braced {
+          w.w(s"return ").nestedN(2) {
+            w.wl(s"\042${self}{\042 +")
+            for (f <- r.fields) {
+              w.wl(s"\042,\042 + \042${idJava.field(f.ident)}=\042 + ${idJava.field(f.ident)} +")
+            }
+          }
+          w.wl(s"\042}\042;")
+        }
```
Comment (mknejp) [inline src/source/JavaGenerator.scala]: You can get rid of the unicode stuff by using triple quotes
`s"""foo"x"foo"""` becomes `foo"x"foo`

Comment (steipete) [inline src/source/JavaGenerator.scala]: I tried that initially, but had troubles with it. Here's what happens when I search/replace for `\042` into `"""`

```
[info] Compiling 4 Scala sources to /Users/steipete/Documents/Projects/djinni/src/target/scala-2.11/classes...
[error] /Users/steipete/Documents/Projects/djinni/src/source/JavaGenerator.scala:347: unclosed string literal
[error]             w.wl(s""""${self}{""" +")
[error]                                    ^
[error] /Users/steipete/Documents/Projects/djinni/src/source/JavaGenerator.scala:349: unclosed string literal
[error]               w.wl(s"""",""" + """${idJava.field(f.ident)}=""" + ${idJava.field(f.ident)} +")
[error]                                                                                            ^
[error] /Users/steipete/Documents/Projects/djinni/src/source/JavaGenerator.scala:352: unclosed string literal
[error]           w.wl(s""""}""";")
[error]                          ^
[error] /Users/steipete/Documents/Projects/djinni/src/source/JavaGenerator.scala:352: ')' expected but ';' found.
[error]           w.wl(s""""}""";")
```

Comment (mknejp) [inline src/source/JavaGenerator.scala]: That's not how it works. The triple quotes enclose the entire string and allow single quotes inside. Check the example again.

Comment (steipete) [inline src/source/JavaGenerator.scala]: Got it! Thanks for taking the time to help me on that!

**Comment by j4cbo** [comment]
Awesome. Same requests as for the ObjC `description` change :)

**Comment by steipete** [comment]
done.

**Pull request merged**